apps/web/src/views/Swap/AkkaSwap/hooks/useAkkaRouterApi.ts

```diff
@@ -28,12 +33,22 @@ export const useAkkaRouterArgs = (token0: Currency, token1: Currency, amount: Cu
       return r.json()
     })
   const { chainId } = useActiveChainId()
-  const API_URL = chainId === ChainId.CORE ? "https://api.akka.foundation" : "https://icecream.akka.finance";
+  const { account } = useWeb3React()
+  const API_URL = chainId === ChainId.CORE ? 'https://api.akka.foundation' : 'https://icecream.akka.finance'
   const { data, error } = useSWR(
-    `${API_URL}/swap?token0=${inputCurrencyId === NATIVE[chainId].symbol ? NATIVE_TOKEN_ADDRESS : token0?.wrapped?.address
-    }&token1=${outputCurrencyId === NATIVE[chainId].symbol ? NATIVE_TOKEN_ADDRESS : token1?.wrapped?.address
-    }&amount=${amount?.multiply(10 ** inputCurrency?.decimals)?.toExact()}&slipage=${slippage / 10000}&use_split=true&${chainId !== ChainId.CORE ? `chain_id=${chainId}` : `chain0=core&chain1=core`}`,
-    token0 && token1 && amount && slippage && (chainId === ChainId.BITGERT || chainId === ChainId.XDC || chainId === ChainId.CORE) && fetcher,
+    `${API_URL}/swap?token0=${
```

Comment:
Maybe its worth it using new URL() to build the url. This way it's way more readable and maintainable